### ISSUE TITLE
Datagen options around missing locales

### ISSUE THREAD

**Issue by sffc** [unicode-org/icu4x#3746]
tl;dr, what should we do when a user tries to export a locale from datagen that isn't in CLDR?

At first thought, it seems that we should inform the client by failing datagen. However, this is more nuanced. Some caveats:

1. Not all keys support the same set of locales.
2. Not all keys have data at the root locale (example: collator/reord@1).
3. Some keys require either an extension (example: datetime/skeletons@1) or soon an auxiliary key.

Because of caveats 2 and 3, we cannot always simply run fallback and fill in the data based on fallback.

@Manishearth suggested tagging data keys that aren't expected to fall back to root with some extra metadata. This fixes caveats 2 and 3 but not 1.

Related questions:

1. Should the behavior depend on whether there was an explicit locale or whether the set of locales came from a CLDR set?
2. Should the behavior depend on the fallback mode (i.e., should Precomputed be stricter than Hybrid)?

CC @robertbastian

**Comment by Manishearth** [comment]
Proposal:

1. Continue printing a warning in datagen when a request language falls back to `und`@ro in an unexpected way
2. Do NOT retain the base language if the language is not in CLDR

LGTM: @robertbastian @sffc 

Discussion:

- @zbraniecki - I think the datagen API should return me a list of locales it failed to generate. OK for the CLI to print warnings.

Conclusion: @sffc/@robertbastian to design an API for this.

**Comment by robertbastian** [comment]
unrelated to missing locales, but I have another use case for `ExportResult`, returning the crates that a baked exporter needs. This is currently only logged.

**Comment by sffc** [comment]
Proposal:

- In 2.0, add non_exhaustive `ExportResult` and return it from datagen driver
- Punt the rest of this issue until we have clearer requirements on the use case

LGTM: @sffc @robertbastian
